### ISSUE TITLE
submodule update fails for swordappv2

### ISSUE THREAD

**Issue by jnugent** [pkp/sword#9]
This is in the stable-3_1_2 branch.

Right now, a fresh checkout of this plugin generates a fatal "reference is not a tree" error:

user@host ~/sword $ git checkout stable-3_1_2
Branch stable-3_1_2 set up to track remote branch stable-3_1_2 from origin.
Switched to a new branch 'stable-3_1_2'
user@host ~/sword $ git submodule update --init --recursive
Submodule 'libs/swordappv2' (https://github.com/quoideneuf/swordappv2-php-library) registered for path 'libs/swordappv2'
Cloning into 'libs/swordappv2'...
remote: Enumerating objects: 609, done.
remote: Total 609 (delta 0), reused 0 (delta 0), pack-reused 609
Receiving objects: 100% (609/609), 2.27 MiB | 908.00 KiB/s, done.
Resolving deltas: 100% (359/359), done.
Checking connectivity... done.
fatal: reference is not a tree: 4193dc6a79a7b3e4af3b5c877b48543d56a95bd4
Unable to checkout '4193dc6a79a7b3e4af3b5c877b48543d56a95bd4' in submodule path 'libs/swordappv2'

Maybe an unpublished commit in the submodule?

Cheers,
Jason

**Comment by asmecher** [comment]
@quoideneuf, is there a chance you deleted a branch or did a force-push or something?

**Comment by quoideneuf** [comment]
Somehow the stable-3_1_2 branch of this repository was looking for a commit that existed in swordapp/swordappv2-php-library but not in quoideneuf/swordappv2-php-library. The commit in question is:

https://github.com/quoideneuf/swordappv2-php-library/commit/4193dc6a79a7b3e4af3b5c877b48543d56a95bd4

That is a merge commit that I didn't have, as it was the result of accepting my changes back to the original repo. I have reset my repo to the origin, so all should be good. 

tldr;
it should work now

**Comment by jnugent** [comment]
Confirmed working, thanks @quoideneuf!

**Comment by asmecher** [comment]
@quoideneuf, so your fork is now equivalent to the original and there's no need for local customization, right? If so, then we can change the submodule URL to the origin repo and there'll be no need for you to keep your own fork (unless you want to)!

**Comment by quoideneuf** [comment]
@asmecher yes, I agree that it can be pointed back to the upstream repo.

**Comment by asmecher** [comment]
Thanks! Merged and back-ported to `stable-3_1_2`.
